src/engine/Server.cpp

```diff
@@ -791,7 +799,10 @@ net::awaitable<Server::PlannedQuery> Server::parseAndPlan(
     SharedCancellationHandle handle, TimeLimit timeLimit) const {
   return computeInNewThread([&query, &qec,
                              enablePatternTrick = enablePatternTrick_,
-                             handle = std::move(handle), timeLimit]() {
+                             handle = std::move(handle), timeLimit]() mutable {
+    // It might take some time until the thread pool is ready,
+    // so reset the state here. Ideally waiting for the thread pool
+    // would periodically check the cancellation state
```

Comment:
If those are the only usages of `computeInNewThread` then maybe you can make the resetting of the cancellation handle part of `computeInNewThread` and to be really safe throw in a `throwIfCancelled` before (you convinced me that they should be two different functions, but still they almost always could appear in pairs).